extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java

```diff
@@ -53,6 +55,27 @@ void build(CombinedIndexBuildItem index,
             BuildProducer<GeneratedBeanBuildItem> generatedBeans,
             BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
 
+        Config config = ConfigProvider.getConfig();
+        Optional<String> springJpaConfig = config.getOptionalValue("spring.jpa.show-sql", String.class);
+        Map<String, String> springJpaPropertiesMap = new HashMap<>();
+        springJpaPropertiesMap.put("spring.jpa.show-sql", "quarkus.hibernate-orm.log.sql");
+        springJpaPropertiesMap.put("spring.jpa.properties.hibernate.dialect ", "quarkus.hibernate-orm.dialect");
+        springJpaPropertiesMap.put("spring.jpa.properties.hibernate.dialect.storage_engine",
+                "quarkus.hibernate-orm.dialect.storage-engine");
+        springJpaPropertiesMap.put("spring.jpa.generate-ddl", "quarkus.hibernate-orm.database.generation");
+
+        Iterable<String> iterablePropertyNames = config.getPropertyNames();
+        List<String> propertyNames = new ArrayList<String>();
+        iterablePropertyNames.forEach(propertyNames::add);
+        List<String> springProperties = propertyNames.stream().filter(s -> s.matches("spring.jpa.*")).collect(toList());
```

Comment:
This regexp is incorrect and should be compiled. It should be `spring\.jpa\..*`. I'm not sure if you want to do that or specifically match the elements of the map.

If you do that, then you need an alternative error message in case the element is not in the map.